bin/fastq_dir_to_samplesheet.py

```diff
@@ -47,35 +118,41 @@ def fastq_dir_to_samplesheet(fastq_dir, samplesheet_file, read1_extension='_R1_0
             os.makedirs(out_dir)
 
         with open(samplesheet_file, "w") as fout:
-            header = ['sample', 'fastq_1', 'fastq_2']
-            fout.write(','.join(header) + '\n')
-            for sample, reads in read_dict.items():
-                sample_info = ','.join([sample] + reads)
-                if len(reads) == 1:
-                    sample_info += ','
-                fout.write(f'{sample_info}\n')
+            header = ["sample", "fastq_1", "fastq_2"]
+            fout.write(",".join(header) + "\n")
+            for sample, reads in sorted(read_dict.items()):
+                for idx, read_1 in enumerate(reads["R1"]):
+                    read_2 = ""
+                    if idx < len(reads["R2"]):
+                        read_2 = reads["R2"][idx]
+                    sample_info = ",".join([sample, read_1, read_2])
+                    fout.write(f"{sample_info}\n")
     else:
-        error_str  = "\nWARNING: No FastQ files found so samplesheet has not been created!\n\n"
+        error_str = (
+            "\nWARNING: No FastQ files found so samplesheet has not been created!\n\n"
+        )
         error_str += "Please check the values provided for the:\n"
         error_str += "  - Path to the directory containing the FastQ files\n"
         error_str += "  - '--read1_extension' parameter\n"
         error_str += "  - '--read2_extension' parameter\n"
         print(error_str)
         sys.exit(1)
 
+
 def main(args=None):
     args = parse_args(args)
 
     fastq_dir_to_samplesheet(
-        fastq_dir               = args.FASTQ_DIR,
-        samplesheet_file        = args.SAMPLESHEET_FILE,
-        read1_extension         = args.READ1_EXTENSION,
-        read2_extension         = args.READ2_EXTENSION,
-        single_end              = args.SINGLE_END,
-        sanitise_name           = args.SANITISE_NAME,
-        sanitise_name_delimiter = args.SANITISE_NAME_DELIMITER,
-        sanitise_name_index     = args.SANITISE_NAME_INDEX
+        fastq_dir=args.FASTQ_DIR,
```

Comment:
Don't want to contradict `black` but I find readable the previous version :stuck_out_tongue_closed_eyes:

Reply:
😏 Me too!